clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/StrongServerStoreProxy.java

```diff
@@ -219,23 +203,15 @@ public Chain get(long key) throws TimeoutException {
 
   @Override
   public void append(final long key, final ByteBuffer payLoad) throws TimeoutException {
-    try {
-      performWaitingForHashInvalidation(key, () -> {
-        delegate.append(key, payLoad);
-        return null;
-      });
-    } catch (InterruptedException ie) {
-      throw new RuntimeException(ie);
-    }
+    performWaitingForHashInvalidation(key, () -> {
+      delegate.append(key, payLoad);
+      return null;
+    }, entity.getTimeouts().getMutativeOperationTimeout().toNanos());
   }
 
   @Override
   public Chain getAndAppend(final long key, final ByteBuffer payLoad) throws TimeoutException {
-    try {
-      return performWaitingForHashInvalidation(key, () -> delegate.getAndAppend(key, payLoad));
-    } catch (InterruptedException ie) {
-      throw new RuntimeException(ie);
-    }
+    return performWaitingForHashInvalidation(key, () -> delegate.getAndAppend(key, payLoad), entity.getTimeouts().getMutativeOperationTimeout().toNanos());
```

Comment:
Why not passing the `TimeoutDuration` in parameter to prevent a redundant `toNanos`?

Reply:
Refactored to be a bit more Java 1.8-ey.